crowd_sim/envs/crowd_sim.py

```diff
@@ -314,6 +385,34 @@ def reset(self, phase='test', test_case=None):
     def onestep_lookahead(self, action):
         return self.step(action, update=False)
 
+    def generate_valid_goal(self, gx, gy, r):
+        while True:
+            collide = False
+            for agent in [self.robot] + self.humans + self.obs:
+                if norm((gx - agent.gx, gy - agent.gy)) < r + agent.radius + self.discomfort_dist:
+                    collide = True
+                    break
+            if not collide:
+                break
+        return gx, gy
+
+    def human_reset_goal(self, human):
+        # eps = 1e-6
+        px, py = human.get_position()
+        vx, vy = human.get_velocity()
+        gx, gy = human.get_goal_position()
+        # still = False
+        # if vx < eps and vy < eps:
+        #     still = True
+        
+        if human.reached_destination():
+            # new_goalx = (np.random.random() - 0.5) * self.square_width
+            # new_goaly = (np.random.random() - 0.5) * self.square_width
+            # new_goalx, new_goaly = self.generate_valid_goal(new_goalx, new_goaly, human.radius)
```

Comment:
Can also use `generate_agent_goal()` here: 
```suggestion
            gx, gy = self.generate_agent_goal()
```

Reply:
Noted!